api/src/test/kotlin/com/google/prefab/api/AndroidTest.kt

```diff
@@ -46,13 +49,103 @@ class AndroidTest {
         val oldLib = mockk<PrebuiltLibrary>()
         val newLib = mockk<PrebuiltLibrary>()
         every { oldLib.platform } returns old
+        every { oldLib.path } returns Paths.get("libfoo.so")
         every { newLib.platform } returns new
+        every { newLib.path } returns Paths.get("libfoo.so")
         assertTrue(new.canUse(oldLib))
         assertFalse(old.canUse(newLib))
     }
 
     @Test
     fun `STL matching constraints are enforced`() {
+        val cxxShared = Android(Android.Abi.Arm64, 21, Android.Stl.CxxShared)
+        val cxxStatic = Android(Android.Abi.Arm64, 21, Android.Stl.CxxStatic)
+        val gnuShared = Android(Android.Abi.Arm64, 21, Android.Stl.GnustlShared)
+        val none = Android(Android.Abi.Arm64, 21, Android.Stl.None)
+        val system = Android(Android.Abi.Arm64, 21, Android.Stl.System)
+
+        // A shared library using c++_shared.
+        val cxxSharedSharedLib = mockk<PrebuiltLibrary>()
+        every { cxxSharedSharedLib.platform } returns cxxShared
+        every { cxxSharedSharedLib.path } returns Paths.get("libfoo.so")
+
+        // A static library using c++_shared.
+        val cxxSharedStaticLib = mockk<PrebuiltLibrary>()
```

Comment:
Is there a copy-paste error with the comment here? There's another block below with "A static library using c++_shared"

Reply:
Yep: https://github.com/google/prefab/pull/18

Reply:
I think this block is OK, and it's the later block that's wrong.